File: Leetcode/CS/Leetcode1423-可获得的最大点数.cs
public class Solution {
    public int MaxScore(int[] cardPoints, int k) {
        int n = cardPoints.Length;
        int windowsize = n - k;
        int sum = 0;
        for(int i = 0;i < windowsize;i++){
            sum += cardPoints[i];
        }
        int minsum = sum;
        for(int i = windowsize;i < n;i++){
            sum += cardPoints[i] - cardPoints[i - windowsize];
            minsum = minsum < sum ? minsum:sum;
        }
        int total = cardPoints.Sum();
        return total - minsum;
    }
}

File: Leetcode/Rust/Leetcode1423-可获得的最大点数.rs
impl Solution {
    pub fn max_score(card_points: Vec<i32>, k: i32) -> i32 {
        let n:usize = card_points.len();
        let windowsize:usize = n - k as usize;
        let mut sum:i32 = 0;
        for i in 0..windowsize{
            sum += card_points[i];
        }
        let mut minsum = sum;
        for i in windowsize..n{
            sum += card_points[i] - card_points[i-windowsize];
            minsum = minsum.min(sum);
        }
        let mut total = 0;
        for i in card_points.iter(){
            total += i;
        }
        total - minsum
    }
}